### PR TITLE
Add Support for Zoom in the Browser

### PR DESCRIPTION
**Pull request by cycomachead**
Alright, I totally thought I'd get to this a month ago...
But I haven't forgotten and I've been using this somewhat regularly for teaching.

Basically, this duplicates a decent bit of code to adapt the zoom.us webpage.
* So far, I can't get the chat to not show up behind the black output section,
so you will need to manually use "Pop Out Chat" and it works well there.
* It is not very graceful when you from being in a waiting room to being admitted to the meeting,
but I don't think this is common if people are logged in. (Refreshing works fine, and you don't need to be re-admitted to the meeting.)

Other new features:
* There is a new setting to hide names of users. I added this because I'll have students who by default will use real names and I'd prefer it not to be in the recording. It defaults to on.
   * I think this needs a bit of work with the defaults though. :/
   * I also realize it's somewhat confusing with the "show first name" setting.
   * Maybe a radio control for names makes more sense?
* There's some CSS / place for logic to adjust the text positioning if there is no profile pic.
   * I will probably want to add a setting to turn this off. Zoom only started showing pfps in the last day.

Otherwise, I've not made many attempts at defining a common interface between the tools. Certainly a better `renderComment` type function can be exposed and shared, as well as the logic for loading settings.

I am happy to have others contribute to my branch, too! :)

### Screenshots

Screen Shot 2021-02-03 at 12 39 28 AM
Screen Shot 2021-02-03 at 12 15 55 AM

### CODE REVIEW

File: youtube.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var showOnlyFirstName;
 
 var highlightWords = [];
 
-$("body").unbind("click").on("click", "yt-live-chat-text-message-renderer,yt-live-chat-paid-message-renderer,yt-live-chat-membership-item-renderer,yt-live-chat-paid-sticker-renderer", function () {
+window.location.href.indexOf('youtube') > 0 && $("body").unbind("click").on("click", "yt-live-chat-text-message-renderer,yt-live-chat-paid-message-renderer,yt-live-chat-membership-item-renderer,yt-live-chat-paid-sticker-renderer", function () {
```
Comment (cycomachead) [inline youtube.js:5]: This is the hack that makes this work on multiple URLs. It's cheesy but seems to work OK. 

If a few things were factored out, it wouldn't be unreasonable to put the entire youtube.js inside a conditional.

File: zoom.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
+/**
+
+<div role="presentation" class="chat-item__chat-info">
```
Comment (cycomachead) [inline zoom.js:3]: This structure is now quite out of date. Should probably delete or update.

File: zoom.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+  $(".hl-c-cont")
+    .addClass("fadeout")
+    .delay(300)
+    .queue(function () {
+      $(".hl-c-cont").remove().dequeue();
+    });
+});
+
+
+function renderChatMessage(message, name, image, opts ) {
```
Comment (cycomachead) [inline zoom.js:145]: This fell behind any recent changes made, but I this was an attempt to define a function that could live elsewhere.

**Comment by cycomachead** [comment]
Here's a quick scaffold of what using a base.js might look like: https://github.com/cycomachead/live-chat-overlay/pull/new/shared-base-zoom
This doesn't yet touch anything YouTube related.

**Comment by champagj** [comment]
> Here's a quick scaffold of what using a base.js might look like: https://github.com/cycomachead/live-chat-overlay/pull/new/shared-base-zoom
> This doesn't yet touch anything YouTube related.

Glad to see someone working to expand Aaron extension to work for Zoom. I would gladly test this if needed let me know if I can help. JF

**Comment by cycomachead** [comment]
The main thing I need help with is improving the mechanism for wrapping the Zoom application. It would be great if there was a to fully restrict the height of everything in Zoom. I think potentially applying some more CSS to `.main` and moving around where the `highlight-comment` element is added would work. 

Anyone can pull my branch and locally install the extension. I've been using it for a few weeks and it is working well enough to use it. (However, it is not nice enough to use as your primary means of running a meeting, but it should work just fine as a second device.)

**Comment by champagj** [comment]
Unfortunately I cannot help with programming. My basic skills allows me to understand structure and do simple mods.

But you may want to look at ZoomOSC. They are developing a custom client that the pros can use for integration with broadcast tools (amongst other things). I am on their beta and nothing about chat but maybe they could built a specific chat box that would work for this application. 
Aaron who did the original has been presenting to the group involved with the beta.

**Comment by cycomachead** [comment]
You mean Alex Lindsay's Office Hours Discord? Yeah, I'm in there.

Right now I don't think Zoom OSC is the right tool for this. We primarily need a display environment that's customizable. ZoomOSC could potentially give us the ability to have a pseduo-API for getting comments out of Zoom, but we'd need a whole lot more to be able to then route them to a displayable format. Plus, my understanding is that OSC is just a command API format, and without "push" functionality, we'd really need to build something quite different.

**Comment by aaronpk** [comment]
Closing in favor of #17

**Pull request closed without merging**